Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -1,3 +1,11 @@
+// Custom NetworkReader that doesn't use C#'s built in MemoryStream in order to
+// avoid allocations.
+//
+// Benchmark: 100kb byte[] passed to NetworkReader constructor 1000x
```

Comment:
did you rerun it once you converted to ArraySegment<byte> ?

Reply:
yes, no difference